app/src/pages/[resourceType]/[id].tsx

```diff
@@ -79,41 +121,66 @@ export default function ResourceIDPage({ jsonData }: InferGetServerSidePropsType
-              <Prism language={'cql' as any} colorScheme="light">
-                {/* eslint-enable  @typescript-eslint/no-explicit-any */}
-                {decodedCql}
-              </Prism>
-            </Tabs.Panel>
-          )}
-          {decodedElm != null && (
-            <Tabs.Panel value="elm" pt="xs">
+        <div>
+          <Tabs variant="outline" defaultValue="json">
+            <Tabs.List>
+              <Tabs.Tab value="json">JSON</Tabs.Tab>
+              {decodedElm != null && <Tabs.Tab value="elm">ELM</Tabs.Tab>}
+              {decodedCql != null && <Tabs.Tab value="cql">CQL</Tabs.Tab>}
+              {jsonData.text && <Tabs.Tab value="narrative">Narrative</Tabs.Tab>}
+              {dataRequirements?.Library.resourceType == 'Library' && (
+                <Tabs.Tab value="datarequirements">Data Requirements</Tabs.Tab>
+              )}
+            </Tabs.List>
+            <Tabs.Panel value="json" pt="xs">
               <Prism language="json" colorScheme="light">
-                {decodedElm}
+                {JSON.stringify(jsonData, null, 2)}
               </Prism>
             </Tabs.Panel>
-          )}
-          {jsonData.text && (
-            <Tabs.Panel value="narrative">
-              <Space h="sm" />
-              {parse(jsonData.text.div)}
-            </Tabs.Panel>
-          )}
-        </Tabs>
+            {decodedCql != null && (
+              <Tabs.Panel value="cql" pt="xs">
+                {/* eslint-disable  @typescript-eslint/no-explicit-any */}
+                <Prism language={'cql' as any} colorScheme="light">
+                  {/* eslint-enable  @typescript-eslint/no-explicit-any */}
+                  {decodedCql}
+                </Prism>
+              </Tabs.Panel>
+            )}
+            {decodedElm != null && (
+              <Tabs.Panel value="elm" pt="xs">
+                <Prism language="json" colorScheme="light">
+                  {decodedElm}
+                </Prism>
+              </Tabs.Panel>
+            )}
+            {jsonData.text && (
+              <Tabs.Panel value="narrative">
+                <Space h="sm" />
+                {parse(jsonData.text.div)}
+              </Tabs.Panel>
+            )}
+            {dataRequirements?.Library.resourceType == 'Library' && (
+              <Tabs.Panel value="datarequirements">
+                <Prism language="json" colorScheme="light">
+                  {JSON.stringify(dataRequirements, null, 2)}
+                </Prism>
+              </Tabs.Panel>
+            )}
+          </Tabs>
+        </div>
+        <div style={{ position: 'absolute', left: '85vw', top: '16vh' }}>
+          <Button
+            id="btn"
+            loading={loadingIconVisible}
+            loaderPosition="center"
+            onClick={() => {
+              refetch();
+              setLoadingIconVisible(true);
+            }}
+          >
+            Get Data Requirements
+          </Button>
```

Comment:
Thats a great idea and a good jumping off point! I think this may be because I am on my smaller monitor, but I think it actually looks better for the buttons to be side-by-side than stacked on top of one another. However I don't have my bigger monitor on me right now, so please let me know if the side-by-side doesn't look that great and should be modified for a varying screen size.

Reply:
They look good side-by-side on my laptop and on my monitor 👍 looks better than having them stacked in my opinion